webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -221,6 +222,7 @@ private void completeResponseFuture(Throwable throwable) {
             responseFuture.complete(this);
         } else {
             LOGGER.finer(() -> log("Response completion failed %s", throwable));
+            Optional.ofNullable(subscription).ifPresent(Flow.Subscription::cancel);
```

Comment:
The use of optional feels overkill here ;)
Why not just `if (subscripton != null) subscription.cancel()`

Reply:
Hah thx fixed